core/src/main/scala/caliban/parsing/VariablesCoercer.scala

```diff
@@ -23,64 +25,77 @@ object VariablesCoercer {
     doc: Document,
     rootType: RootType,
     skipValidation: Boolean
-  )(implicit trace: Trace): IO[ValidationError, GraphQLRequest] = {
+  )(implicit trace: Trace): IO[ValidationError, GraphQLRequest] =
+    coerceVariables(req.variables.getOrElse(Map.empty), doc, rootType, skipValidation)
+      .map(m => req.copy(variables = Some(m)))
+
+  def coerceVariables(
+    variables: Map[String, InputValue],
+    doc: Document,
+    rootType: RootType,
+    skipValidation: Boolean
+  )(implicit trace: Trace): IO[ValidationError, Map[String, InputValue]] = {
```

Comment:
Not necessary as an optimization, but I found it a bit weird that we need to pass the `GraphQLRequest` to this method only to extract the variables and copy them back in.

Do we see a case in the future where we might need more information from the `GraphQLRequest` case class? Happy to rollback this change if we don't want to add this overloaded method

Reply:
I think what you did is okay